core/src/main/java/com/linecorp/armeria/server/cors/AbstractCorsPolicyBuilder.java

```diff
@@ -121,27 +121,8 @@ B setConfig(CorsDecorator corsDecorator) {
      * @return {@code this} to support method chaining.
      * @throws IllegalArgumentException if the path pattern is not valid
      */
-    public B pathMapping(String pathPattern) {
-        return pathMapping(PathMapping.of(pathPattern));
-    }
-
-    /**
-     * Adds a {@link PathMapping} that this policy is supposed to be applied to.
-     *
-     * @param pathMapping the {@link PathMapping} that this policy is supposed to be applied to
-     * @return {@code this} to support method chaining.
-     * @throws IllegalArgumentException if the {@link PathMapping} has conditions beyond the path pattern,
-     *                                  i.e. the {@link PathMapping} created by
-     *                                  {@link PathMapping#withHttpHeaderInfo(Set, List, List)}
-     */
-    public B pathMapping(PathMapping pathMapping) {
-        requireNonNull(pathMapping, "pathMapping");
-        if (!pathMapping.hasPathPatternOnly()) {
-            throw new IllegalArgumentException(
-                    "pathMapping: " + pathMapping.getClass().getSimpleName() +
-                    " (expected: the path mapping which has only the path patterns as its condition)");
-        }
-        pathMappings.add(pathMapping);
+    public B route(String pathPattern) {
```

Comment:
Don't necessary to add `route(Route route)`?

Reply:
I didn't add the API intentionally because the user should use just the path. In the previous implementation, we were calling `pathMapping.hasPathPatternOnly()` to check whether the `pathMapping` has conditions other than that. So I thought just these APIs enough for the users because they do not have to specify HTTP methods, consumes and produces.
But I'm not so strong on this and if we still need the API I can add it.

Reply:
+1 for not accepting `Route`

Reply:
I mis-thought why I made the limitation which requires path pattern only. Yes, you are right. It's still not useful after this changes. I mis-thought this change can overcome the limitation . 😅